### PR TITLE
feat(platform-server): update domino to v2.1.0

### PR DESCRIPTION
**Pull request by CaerusKaru**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/core/test/render3/compiler_canonical/elements_spec.ts [hunk 1]
```diff
@@ -311,7 +311,11 @@ describe('elements', () => {
       }
 
       const comp = renderComponent(MyComponent);
-      expect(toHtml(comp)).toEqual('<div></div>');
+
+      // This is a fix for a change in how Domino renders this on the server in v2.1.0
+      const source = toHtml(comp);
+      const matches = source === '<div></div>' || source === '<div class=""></div>';
```
Comment (IgorMinar) [inline packages/core/test/render3/compiler_canonical/elements_spec.ts]: this is weird. why would it serialize an empty string attribute? are we doing something that triggers this behavior?

my concern is that we'll be bloating the html size if we happen to do this on every element.

**Review by IgorMinar — COMMENTED** [review]
other than the `class=""` this looks good to me.

File: packages/core/test/render3/compiler_canonical/elements_spec.ts (hunk 1 above)
Comment (gkalpak) [reply packages/core/test/render3/compiler_canonical/elements_spec.ts]: I don't know about domino, but some attributes affect the element behavior even if empty.

The `||` is weird though. Are the tests run against multiple versions of Domino?

File: packages/platform-server/test/integration_spec.ts [hunk 2]
```diff
@@ -588,7 +588,7 @@ class HiddenModule {
              expect(output).toContain('Works!');
              expect(output).toContain('ng-trigger-myAnimation');
              expect(output).toContain('opacity:1;');
-             expect(output).toContain('transform:translate3d(0, 0, 0);');
+             expect(output).toContain('transform:translate3d(0 , 0 , 0);');
```
Comment (skreborn) [inline packages/platform-server/test/integration_spec.ts]: Why did you add extra spaces here? This seems like an accident.

Comment (CaerusKaru) [reply packages/platform-server/test/integration_spec.ts]: It’s not an accident, Domino changed how it serializes that particular form of CSS

File: packages/core/test/render3/compiler_canonical/elements_spec.ts (hunk 1 above)
Comment (CaerusKaru) [reply packages/core/test/render3/compiler_canonical/elements_spec.ts]: The tests are run against Domino and against the browser. The browser does not exhibit this behavior, but Domino does, so the test needs to accept both.

Comment (IgorMinar) [reply packages/core/test/render3/compiler_canonical/elements_spec.ts]: We should understand why domino does this. Emitting these extra classes will cause HTML bloat which is undesirable.

Does this happen only on some elements or all elements? is this a bug in domino that we should report and fix?

I don't feel comfortable forcing people to update to domino 2.1 if it causes these kinds of issues, so before we do so, we need to understand the severity of this issue.

Comment (CaerusKaru) [reply packages/core/test/render3/compiler_canonical/elements_spec.ts]: The class issue is only on this test because it does the following: create a div, attach a conditional class based on a boolean for the component (initialized to false), then set the boolean to true. In every other test, there is no bloat, but the act of initializing an empty class in Domino results in an empty string on the div.

**Review by IgorMinar — APPROVED** [review]
I discussed this with Vikram and the change looks good to me. The empty attribute issue doesn't seem to be widespread enough to warrant blocking this.

**Review by IgorMinar — APPROVED** [review]
I discussed this with Vikram and the change looks good to me. The empty attribute issue doesn't seem to be widespread enough to warrant blocking this.

This PR however needs to be rebased before we can merge it.

**Review by vikerman — APPROVED** [review]
